laufband/laufband.py

```diff
@@ -163,114 +167,65 @@ def close(self):
         Instead, you can use this function to exit
         the laufband generator marking the job as completed.
         """
-        self._close_trigger = True
+        self._graphband.close()
 
     @property
     def identifier(self) -> str:
         """Unique identifier of this worker"""
-        return self.db.worker
+        return self._graphband.identifier
+
+    @property
+    def lock(self):
+        """Access to the underlying lock"""
+        return self._graphband.lock
+
+    @property
+    def disabled(self) -> bool:
+        """Whether Laufband is disabled"""
+        return self._graphband.disabled
+
+    @property
+    def com(self):
+        """Path to the communication/database file"""
+        return self._graphband.com
 
     @property
-    @_check_disabled
     def completed(self) -> list[int]:
         """Return the indices of items that have been completed."""
-        with self.lock:
-            return self.db.list_state("completed")
+        # Convert string task IDs back to integers for backwards compatibility
+        task_ids = self._graphband.completed
+        return [int(task_id) for task_id in task_ids if task_id.isdigit()]
 
     @property
-    @_check_disabled
     def failed(self) -> list[int]:
         """Return the indices of items that have failed processing."""
-        with self.lock:
-            return self.db.list_state("failed")
+        task_ids = self._graphband.failed
+        return [int(task_id) for task_id in task_ids if task_id.isdigit()]
 
     @property
-    @_check_disabled
     def running(self) -> list[int]:
         """Return the indices of items that are currently being processed."""
-        with self.lock:
-            return self.db.list_state("running")
+        task_ids = self._graphband.running
+        return [int(task_id) for task_id in task_ids if task_id.isdigit()]
 
     @property
-    @_check_disabled
     def pending(self) -> list[int]:
         """Return the indices of items that are pending processing."""
-        with self.lock:
-            return self.db.list_state("pending")
+        task_ids = self._graphband.pending
+        return [int(task_id) for task_id in task_ids if task_id.isdigit()]
 
     @property
-    @_check_disabled
     def died(self) -> list[int]:
         """Return the indices of items that have been marked as 'died'."""
-        with self.lock:
-            return self.db.list_state("died")
+        task_ids = self._graphband.died
+        return [int(task_id) for task_id in task_ids if task_id.isdigit()]
```

Comment:
it is nice, that all of these can still return `list[int]` but what if I'd pass a hash_fn?